pennylane/operation.py

```diff
@@ -697,8 +703,43 @@ def compute_decomposition(theta, wires):
     # taken from [stackexchange](https://stackoverflow.com/questions/40694380/forcing-multiplication-to-use-rmul-instead-of-numpy-array-mul-or-byp/44634634#44634634)
     __array_priority__ = 1000
 
+    _primitive = None
+    """
+    Optional[jax.core.Primitive]
+    """
+
     def __init_subclass__(cls, **_):
         register_pytree(cls, cls._flatten, cls._unflatten)
+        cls._primitive = create_operator_primitive(cls)
+
+    @classmethod
+    def _primitive_bind_call(cls, *args, **kwargs):
+        """This class method should match the call signature of the class itself.
```

Comment:
Sooo... Should the signature of this implementation match the `__init__` signature
```python
def __init__(self, *params, wires=None, id=None):
```
?

Reply:
I think `(*args, **kwargs)` is a little more versatile and less opinionated.  If we set `(*params, wires=None, id=None)` as the call signature, we would have to override this method for any class with a slightly different call signature, even if it's just the addition of a keyword argument.

By making it as unopinionated as possible, fewer operators and templates have to care about this method.

Reply:
Agreed :)